lib/compile/util.js

```diff
@@ -50,14 +50,15 @@ function checkDataType(dataType, data, negate) {
     case 'integer': return '(typeof ' + data + EQUAL + '"number"' + AND +
                            NOT + '(' + data + ' % 1)' +
                            AND + data + EQUAL + data + ')';
+    case 'number': return '(typeof ' + data + EQUAL + '"' + dataType + '"' +  (strictNumbers ? (AND + OK + 'isFinite(' + data + ')') : '') + ')';
```

Comment:
The reason to use `data === data` is because it's ES5 compatible.
isFinite was only added in ES6 (I think), so it would only be possible if ajv drops ES5 support for generated code...
If I am correct about ES6, then it should be `data === data && data !== Infinity && data !== -Infinity` in the generated code.

Reply:
It appears as though `isFinite` is available in ES5 (I think it's actually been there since the beginning) although it didn't start hanging off the `Number` prototype since ES6. This code should still work in es5. https://www.ecma-international.org/ecma-262/5.1/#sec-15.1.2.5

Reply:
cool, then it can be used (needs some benchmarking, but https://jsbench.me/ shows them as equal)